File: server/model/src/main/kotlin/hu/bsstudio/bssweb/event/model/Event.kt
package hu.bsstudio.bssweb.event.model

import java.time.LocalDate
import java.util.UUID

data class Event(
        val id: UUID,
        val url: String,
        val title: String,
        val description: String,
        val dateFrom: LocalDate,
        val visible: Boolean,
)

File: integration/src/types/event.ts
import type { UUID } from './common.js';

export interface CreateEvent {
  url: string;
  title: string;
}

export interface UpdateEvent extends CreateEvent {
  description: string;
  dateFrom: string; // ISO date string
  dateTo: string; // ISO date string
  visible: boolean;
}

export interface Event extends UpdateEvent {
  id: UUID;
}

export interface DetailedEvent extends Event {
  // Additional fields from DetailedEvent if any
}
